### PR TITLE
Replace try! with ?.

### PR DESCRIPTION
**Pull request by ahmedcharles**

### CODE REVIEW

**Comment by nrc** [comment]
lgtm, but just checking with @rust-lang/compiler that they approve

**Comment by eddyb** [comment]
cc @jonathandturner Does this change anything `libsyntax` uses or might use?
cc @erickt @dtolnay Making sure syntex doesn't get blocked.

**Comment by nrc** [comment]
This is all internal stuff, linbsyntax is touched in https://github.com/rust-lang/rust/pull/36037

**Comment by eddyb** [comment]
@nrc I was asking mostly because of a potential move of libsyntax to the error code system used in rustc.

**Comment by ahmedcharles** [comment]
I can move changes (if there are any) that would impact syntex to #36037, easily enough and then get this merged.

**Comment by dtolnay** [comment]
None of these affect syntex.

**Comment by nikomatsakis** [comment]
👍

**Comment by ahmedcharles** [comment]
I assume this needs to be approved again @nrc?

**Comment by nrc** [comment]
@ahmedcharles yes, and unfortunately it needs another rebase now. When you push new commits, it is a good idea to ping the reviewer in a comment since GitHub does not send notifications about new commits (thus why I missed this).

**Comment by dtolnay** [comment]
@nrc 

> GitHub does not send notifications about new commits (thus why I missed this).

It does if you check the "Pull Request pushes" box at https://github.com/settings/notifications:

selection_007

**Comment by jonathandturner** [comment]
@dtolnay - if we checked that, for the amount of PRs we review, we'd likely swamp our inbox and still not see it, sadly

**Comment by ahmedcharles** [comment]
@nrc Probably ready for another try.

**Comment by ahmedcharles** [comment]
Seems like an intermittent failure @alexcrichton @nrc ?

File: src/librustc_incremental/persist/util.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
```
Comment (TimNN) [inline src/librustc_incremental/persist/util.rs]: This file does not look like it's actually part of this PR.

**Comment by ahmedcharles** [comment]
The merge conflict issue is fixed. @nrc.

**Pull request merged**